### ISSUE TITLE
Cannot add high latency node in microk8s cluster. (i/o timeout)

### ISSUE THREAD

**Issue by ThanKarab** [canonical/microk8s#3688]
#### Summary
I am deploying a k8s cluster with microk8s and 4 nodes. 2 nodes have high latency (~200ms) and the other 2 are in the same cloud provider. I understand that this is not a usual use case but I am not trying to deploy HA cluster with many master nodes, only worker nodes that have high latency.

I am trying 2 different scenarios.
1) A high latency node is the master node and then the rest of the nodes (2 fast, 1 with high latency) join the cluster. This scenario works like a charm.
2) A "fast" node is the master node and then the rest of the nodes join the cluster. This scenario throws i/o timeout errors.

#### What Should Happen Instead?
I would expect both scenarios to be working and somehow to be able to configure how much the timeout is in seconds.

#### Reproduction Steps

1. Create a microk8s v1.26 cluster.
2. Join a "fast" node in the cluster.
3. Join a high latency node in the cluster.

#### Introspection Report

[inspection-report-20230124_133203.tar.gz](https://github.com/canonical/microk8s/files/10490456/inspection-report-20230124_133203.tar.gz)

#### Can you suggest a fix?

From what I understand there is a configuration that is different in the 2 scenarios, I just can't find what that configuration is that allows in some cases for high latency nodes to be accepted.

**Comment by sachinkumarsingh092** [comment]
Hi @ThanKarab, can you see if increasing the timeout helps?
You can do this by including the argument `--node-status-update-frequency` in kubelet. It has a default value of 10s. You can increase it to see if it helps. The current arguments for kubelet can be found at `/var/snap/microk8s/current/args/kubelet`.

**Comment by ThanKarab** [comment]
Hello @sachinkumarsingh092, should I change this argument in all nodes?

**Comment by ThanKarab** [comment]
Didn't add this in the ticket but this is the error I am getting in the `install-cni` of calico:

```
microk8s kubectl logs calico-node-w854w -n kube-system -c install-cni
time="2023-01-24T13:26:36Z" level=info msg="Running as a Kubernetes pod" source="install.go:140"
2023-01-24 13:26:37.165 [INFO][1] cni-installer/<nil> <nil>: File is already up to date, skipping file="/host/opt/cni/bin/bandwidth"
2023-01-24 13:26:37.165 [INFO][1] cni-installer/<nil> <nil>: Installed /host/opt/cni/bin/bandwidth
2023-01-24 13:26:37.235 [INFO][1] cni-installer/<nil> <nil>: File is already up to date, skipping file="/host/opt/cni/bin/calico"
2023-01-24 13:26:37.235 [INFO][1] cni-installer/<nil> <nil>: Installed /host/opt/cni/bin/calico
2023-01-24 13:26:37.286 [INFO][1] cni-installer/<nil> <nil>: File is already up to date, skipping file="/host/opt/cni/bin/calico-ipam"
2023-01-24 13:26:37.286 [INFO][1] cni-installer/<nil> <nil>: Installed /host/opt/cni/bin/calico-ipam
2023-01-24 13:26:37.289 [INFO][1] cni-installer/<nil> <nil>: File is already up to date, skipping file="/host/opt/cni/bin/flannel"
2023-01-24 13:26:37.289 [INFO][1] cni-installer/<nil> <nil>: Installed /host/opt/cni/bin/flannel
2023-01-24 13:26:37.293 [INFO][1] cni-installer/<nil> <nil>: File is already up to date, skipping file="/host/opt/cni/bin/host-local"
2023-01-24 13:26:37.293 [INFO][1] cni-installer/<nil> <nil>: Installed /host/opt/cni/bin/host-local
2023-01-24 13:26:37.349 [INFO][1] cni-installer/<nil> <nil>: File is already up to date, skipping file="/host/opt/cni/bin/install"
2023-01-24 13:26:37.349 [INFO][1] cni-installer/<nil> <nil>: Installed /host/opt/cni/bin/install
2023-01-24 13:26:37.352 [INFO][1] cni-installer/<nil> <nil>: File is already up to date, skipping file="/host/opt/cni/bin/loopback"
2023-01-24 13:26:37.352 [INFO][1] cni-installer/<nil> <nil>: Installed /host/opt/cni/bin/loopback
2023-01-24 13:26:37.355 [INFO][1] cni-installer/<nil> <nil>: File is already up to date, skipping file="/host/opt/cni/bin/portmap"
2023-01-24 13:26:37.355 [INFO][1] cni-installer/<nil> <nil>: Installed /host/opt/cni/bin/portmap
2023-01-24 13:26:37.358 [INFO][1] cni-installer/<nil> <nil>: File is already up to date, skipping file="/host/opt/cni/bin/tuning"
2023-01-24 13:26:37.358 [INFO][1] cni-installer/<nil> <nil>: Installed /host/opt/cni/bin/tuning
2023-01-24 13:26:37.358 [INFO][1] cni-installer/<nil> <nil>: Wrote Calico CNI binaries to /host/opt/cni/bin

2023-01-24 13:26:37.400 [INFO][1] cni-installer/<nil> <nil>: CNI plugin version: v3.23.5

2023-01-24 13:26:37.400 [INFO][1] cni-installer/<nil> <nil>: /host/secondary-bin-dir is not writeable, skipping
W0124 13:26:37.400648       1 client_config.go:617] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
2023-01-24 13:27:07.402 [ERROR][1] cni-installer/<nil> <nil>: Unable to create token for CNI kubeconfig error=Post "https://10.152.183.1:443/api/v1/namespaces/kube-system/serviceaccounts/calico-node/token": dial tcp 10.152.183.1:443: i/o timeout
2023-01-24 13:27:07.402 [FATAL][1] cni-installer/<nil> <nil>: Unable to create token for CNI kubeconfig error=Post "https://10.152.183.1:443/api/v1/namespaces/kube-system/serviceaccounts/calico-node/token": dial tcp 10.152.183.1:443: i/o timeout
```

**Comment by neoaggelos** [comment]
Hi @ThanKarab

In the inspection report, I see:

```
NAMESPACE            NAME                                            READY   STATUS                  RESTARTS      AGE     IP                NODE         NOMINATED NODE   READINESS GATES
kube-system          pod/calico-node-9vxwr                           1/1     Running                 0             14m     10.86.14.11       sandboxwk1   <none>           <none>
kube-system          pod/calico-node-s7bwj                           1/1     Running                 0             6m34s   10.86.14.12       sandboxwk2   <none>           <none>
kube-system          pod/calico-node-dj2gq                           0/1     Init:CrashLoopBackOff   5 (32s ago)   6m44s   206.189.201.218   mipnyc1      <none>           <none>
```

By the looks of it, I don't think the issue is the high latency. Are all your nodes connected to the same network?

Please, on the node you have installed run:

```
ip route
sudo microk8s kubectl get node -o wide
sudo microk8s kubectl get endpoints kubernetes
```

And on the second node (where calico-node fails) run:

```
ip route
```

`10.152.183.1` is a virtual IP address assigned by Kubernetes that forwards all traffic to the Kubernetes endpoints. If your nodes cannot access that IP (e.g. possible if behind NAT) then this could explain why you see i/o timeouts.

There are ways to solve this problem, but can you first confirm whether this is actually the issue? (perhaps i am reading too much into this and that is not the case). Thanks!

**Comment by ThanKarab** [comment]
Hello @neoaggelos  and thank you very much for the assistance.

On the master node:
```
> ip route
default via 10.86.13.1 dev ens3 proto dhcp src 10.86.13.10 metric 100
10.1.107.64/26 via 10.1.107.64 dev vxlan.calico onlink
blackhole 10.1.146.128/26 proto 80
10.1.146.138 dev calid7fd8e91297 scope link
10.1.146.139 dev cali506fd7f612d scope link
10.1.146.140 dev cali530e5f2bf80 scope link
10.1.146.141 dev calif5bd0a9b562 scope link
10.1.146.142 dev calif904f69f571 scope link
10.1.146.143 dev calibe396168c47 scope link
10.1.146.144 dev calie72463e3879 scope link
10.1.146.145 dev cali767fe06b9b1 scope link
10.1.229.128/26 via 10.1.229.128 dev vxlan.calico onlink
10.86.13.0/24 dev ens3 proto kernel scope link src 10.86.13.10 metric 100
10.86.13.1 dev ens3 proto dhcp scope link src 10.86.13.10 metric 100
10.86.13.100 dev ens3 proto dhcp scope link src 10.86.13.10 metric 100
10.86.14.0/24 dev tap1 proto kernel scope link src 10.86.14.10
10.86.20.2 via 10.86.13.1 dev ens3 proto dhcp src 10.86.13.10 metric 100
169.254.169.254 via 10.86.13.100 dev ens3 proto dhcp src 10.86.13.10 metric 100
>microk8s kubectl get node -o wide
NAME         STATUS   ROLES    AGE     VERSION   INTERNAL-IP       EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION      CONTAINER-RUNTIME
mipnyc1      Ready    <none>   5h39m   v1.26.0   206.189.201.218   <none>        Ubuntu 22.04.1 LTS   5.15.0-50-generic   containerd://1.6.8
sandboxms    Ready    <none>   8h      v1.26.0   10.86.14.10       <none>        Ubuntu 22.04.1 LTS   5.15.0-53-generic   containerd://1.6.8
sandboxwk1   Ready    <none>   8h      v1.26.0   10.86.14.11       <none>        Ubuntu 22.04.1 LTS   5.15.0-53-generic   containerd://1.6.8
> microk8s kubectl get endpoints kubernetes
NAME         ENDPOINTS           AGE
kubernetes   10.86.13.10:16443   8h
```

On the node where calico-node fails:
```
> ip route
default via 206.189.192.1 dev eth0 proto static
10.1.0.0/16 via 100.64.0.1 dev ovn0
blackhole 10.1.107.64/26 proto 80
10.1.107.66 dev cali5089b2798cb scope link
10.1.146.128/26 via 10.1.146.128 dev vxlan.calico onlink
10.1.229.128/26 via 10.1.229.128 dev vxlan.calico onlink
10.10.0.0/16 dev eth0 proto kernel scope link src 10.10.0.5
10.86.14.0/24 dev tap1 proto kernel scope link src 10.86.14.14
10.86.20.0/24 via 10.86.14.1 dev tap1
10.116.0.0/20 dev eth1 proto kernel scope link src 10.116.0.2
100.64.0.0/16 dev ovn0 proto kernel scope link src 100.64.0.4
206.189.192.0/20 dev eth0 proto kernel scope link src 206.189.201.218

```

I am not really sure how the `10.152.183.1` address comes up...

**Comment by ThanKarab** [comment]
This is from another execution where calico starts but a little bit later it fails:
```
2023-01-25 16:11:29.359 [INFO][93] monitor-addresses/reachaddr.go 47: Auto-detected address by connecting to remote Destination="10.86.14.11" IP=10.86.14.14                                                                                                       [1569/36710]
2023-01-25 16:11:29.361 [INFO][93] monitor-addresses/autodetection_methods.go 143: Using autodetected IPv4 address 10.86.14.14/24, detected by connecting to 10.86.14.11
2023-01-25 16:11:56.337 [INFO][96] felix/vxlan_resolver.go 328: VTEP information has not changed, skipping duplicate update node="sandboxms"
2023-01-25 16:11:56.338 [INFO][96] felix/vxlan_resolver.go 328: VTEP information has not changed, skipping duplicate update node="sandboxms"
2023-01-25 16:12:14.795 [INFO][96] felix/summary.go 100: Summarising 11 dataplane reconciliation loops over 1m6.4s: avg=5ms longest=23ms ()
2023-01-25 16:12:29.362 [INFO][93] monitor-addresses/reachaddr.go 47: Auto-detected address by connecting to remote Destination="10.86.14.11" IP=10.86.14.14
2023-01-25 16:12:29.363 [INFO][93] monitor-addresses/autodetection_methods.go 143: Using autodetected IPv4 address 10.86.14.14/24, detected by connecting to 10.86.14.11
2023-01-25 16:13:17.258 [INFO][96] felix/summary.go 100: Summarising 10 dataplane reconciliation loops over 1m2.5s: avg=7ms longest=13ms (resync-ipsets-v4)
2023-01-25 16:13:29.364 [INFO][93] monitor-addresses/reachaddr.go 47: Auto-detected address by connecting to remote Destination="10.86.14.11" IP=10.86.14.14
2023-01-25 16:13:29.366 [INFO][93] monitor-addresses/autodetection_methods.go 143: Using autodetected IPv4 address 10.86.14.14/24, detected by connecting to 10.86.14.11
2023-01-25 16:14:20.579 [INFO][96] felix/iface_monitor.go 217: Netlink address update for known interface.  addr="10.86.14.14" exists=false ifIndex=28
2023-01-25 16:14:20.579 [INFO][96] felix/int_dataplane.go 1198: Linux interface addrs changed. addrs=set.mapSet{fe80::4438:eeff:fefd:e688} ifaceName="tap1"                                                                                                                    2023-01-25 16:14:20.580 [INFO][96] felix/int_dataplane.go 1618: Received interface addresses update msg=&intdataplane.ifaceAddrsUpdate{Name:"tap1", Addrs:set.mapSet{"fe80::4438:eeff:fefd:e688":set.empty{}}}
2023-01-25 16:14:20.580 [INFO][96] felix/hostip_mgr.go 85: Interface addrs changed. update=&intdataplane.ifaceAddrsUpdate{Name:"tap1", Addrs:set.mapSet{"fe80::4438:eeff:fefd:e688":set.empty{}}}
2023-01-25 16:14:20.580 [INFO][96] felix/ipsets.go 130: Queueing IP set for creation family="inet" setID="this-host" setType="hash:ip"
2023-01-25 16:14:20.581 [INFO][96] felix/ipsets.go 785: Doing full IP set rewrite family="inet" numMembersInPendingReplace=7 setID="this-host"
2023-01-25 16:14:20.601 [INFO][96] felix/summary.go 100: Summarising 12 dataplane reconciliation loops over 1m3.3s: avg=9ms longest=38ms ()
2023-01-25 16:14:23.744 [WARNING][96] felix/vxlan_mgr.go 534: Failed to configure VXLAN tunnel device, retrying... error=Unable to find parent interface with address 10.86.14.14 ipVersion=0x4
2023-01-25 16:14:24.745 [WARNING][96] felix/vxlan_mgr.go 534: Failed to configure VXLAN tunnel device, retrying... error=Unable to find parent interface with address 10.86.14.14 ipVersion=0x4                                                                                2023-01-25 16:14:25.749 [WARNING][96] felix/vxlan_mgr.go 534: Failed to configure VXLAN tunnel device, retrying... error=Unable to find parent interface with address 10.86.14.14 ipVersion=0x4
2023-01-25 16:14:26.750 [WARNING][96] felix/vxlan_mgr.go 534: Failed to configure VXLAN tunnel device, retrying... error=Unable to find parent interface with address 10.86.14.14 ipVersion=0x4
2023-01-25 16:14:27.752 [WARNING][96] felix/vxlan_mgr.go 534: Failed to configure VXLAN tunnel device, retrying... error=Unable to find parent interface with address 10.86.14.14 ipVersion=0x4
2023-01-25 16:14:28.754 [WARNING][96] felix/vxlan_mgr.go 534: Failed to configure VXLAN tunnel device, retrying... error=Unable to find parent interface with address 10.86.14.14 ipVersion=0x4
2023-01-25 16:14:29.366 [INFO][93] monitor-addresses/reachaddr.go 47: Auto-detected address by connecting to remote Destination="10.86.14.11" IP=206.189.201.218
2023-01-25 16:14:29.368 [INFO][93] monitor-addresses/autodetection_methods.go 143: Using autodetected IPv4 address 206.189.201.218/20, detected by connecting to 10.86.14.11
2023-01-25 16:14:29.368 [INFO][93] monitor-addresses/startup.go 561: Node IPv4 changed, will check for conflicts
2023-01-25 16:14:29.756 [WARNING][96] felix/vxlan_mgr.go 534: Failed to configure VXLAN tunnel device, retrying... error=Unable to find parent interface with address 10.86.14.14 ipVersion=0x4
2023-01-25 16:14:30.757 [WARNING][96] felix/vxlan_mgr.go 534: Failed to configure VXLAN tunnel device, retrying... error=Unable to find parent interface with address 10.86.14.14 ipVersion=0x4
2023-01-25 16:14:31.759 [WARNING][96] felix/vxlan_mgr.go 534: Failed to configure VXLAN tunnel device, retrying... error=Unable to find parent interface with address 10.86.14.14 ipVersion=0x4
2023-01-25 16:14:32.489 [INFO][96] felix/iface_monitor.go 217: Netlink address update for known interface.  addr="10.86.14.14" exists=true ifIndex=28
2023-01-25 16:14:32.489 [INFO][96] felix/int_dataplane.go 1198: Linux interface addrs changed. addrs=set.mapSet{10.86.14.14,fe80::4438:eeff:fefd:e688} ifaceName="tap1"                                                                                                        2023-01-25 16:14:32.489 [INFO][96] felix/int_dataplane.go 1618: Received interface addresses update msg=&intdataplane.ifaceAddrsUpdate{Name:"tap1", Addrs:set.mapSet{"10.86.14.14":set.empty{}, "fe80::4438:eeff:fefd:e688":set.empty{}}}
2023-01-25 16:14:32.489 [INFO][96] felix/hostip_mgr.go 85: Interface addrs changed. update=&intdataplane.ifaceAddrsUpdate{Name:"tap1", Addrs:set.mapSet{"10.86.14.14":set.empty{}, "fe80::4438:eeff:fefd:e688":set.empty{}}}
2023-01-25 16:14:32.490 [INFO][96] felix/ipsets.go 130: Queueing IP set for creation family="inet" setID="this-host" setType="hash:ip"
2023-01-25 16:14:32.491 [INFO][96] felix/ipsets.go 785: Doing full IP set rewrite family="inet" numMembersInPendingReplace=8 setID="this-host"
2023-01-25 16:14:34.276 [INFO][96] felix/vxlan_resolver.go 328: VTEP information has not changed, skipping duplicate update node="sandboxwk1"
2023-01-25 16:14:34.276 [INFO][96] felix/vxlan_resolver.go 328: VTEP information has not changed, skipping duplicate update node="sandboxwk1"
2023-01-25 16:14:37.365 [INFO][96] felix/calc_graph.go 460: Local endpoint updated id=WorkloadEndpoint(node=mipnyc1, orchestrator=k8s, workload=ingress/nginx-ingress-microk8s-controller-7th8s, name=eth0)                                                                    2023-01-25 16:14:37.366 [INFO][96] felix/int_dataplane.go 1585: Received *proto.WorkloadEndpointUpdate update from calculation graph msg=id:<orchestrator_id:"k8s" workload_id:"ingress/nginx-ingress-microk8s-controller-7th8s" endpoint_id:"eth0" > endpoint:<state:"active"
name:"cali5089b2798cb" profile_ids:"kns.ingress" profile_ids:"ksa.ingress.nginx-ingress-microk8s-serviceaccount" ipv4_nets:"10.1.107.66/32" >
2023-01-25 16:14:37.366 [INFO][96] felix/endpoint_mgr.go 607: Updating per-endpoint chains. id=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"ingress/nginx-ingress-microk8s-controller-7th8s", EndpointId:"eth0"}
2023-01-25 16:14:37.366 [INFO][96] felix/table.go 508: Queueing update of chain. chainName="cali-tw-cali5089b2798cb" ipVersion=0x4 table="filter"
2023-01-25 16:14:37.366 [INFO][96] felix/table.go 508: Queueing update of chain. chainName="cali-fw-cali5089b2798cb" ipVersion=0x4 table="filter"
2023-01-25 16:14:37.366 [INFO][96] felix/endpoint_mgr.go 653: Updating endpoint routes. id=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"ingress/nginx-ingress-microk8s-controller-7th8s", EndpointId:"eth0"}
2023-01-25 16:14:37.366 [INFO][96] felix/endpoint_mgr.go 1220: Applying /proc/sys configuration to interface. ifaceName="cali5089b2798cb"
2023-01-25 16:14:37.366 [INFO][96] felix/endpoint_mgr.go 495: Re-evaluated workload endpoint status adminUp=true failed=false known=true operUp=true status="up" workloadEndpointID=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"ingress/nginx-ingress-microk8s-c
ontroller-7th8s", EndpointId:"eth0"}
2023-01-25 16:14:37.366 [INFO][96] felix/status_combiner.go 58: Storing endpoint status update ipVersion=0x4 status="up" workload=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"ingress/nginx-ingress-microk8s-controller-7th8s", EndpointId:"eth0"}
2023-01-25 16:14:37.373 [INFO][96] felix/status_combiner.go 81: Endpoint up for at least one IP version id=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"ingress/nginx-ingress-microk8s-controller-7th8s", EndpointId:"eth0"} ipVersion=0x4 status="up"
2023-01-25 16:14:37.373 [INFO][96] felix/status_combiner.go 98: Reporting combined status. id=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"ingress/nginx-ingress-microk8s-controller-7th8s", EndpointId:"eth0"} status="up"
2023-01-25 16:14:58.105 [INFO][96] felix/calc_graph.go 460: Local endpoint updated id=WorkloadEndpoint(node=mipnyc1, orchestrator=k8s, workload=ingress/nginx-ingress-microk8s-controller-7th8s, name=eth0)
2023-01-25 16:14:58.105 [INFO][96] felix/int_dataplane.go 1585: Received *proto.WorkloadEndpointUpdate update from calculation graph msg=id:<orchestrator_id:"k8s" workload_id:"ingress/nginx-ingress-microk8s-controller-7th8s" endpoint_id:"eth0" > endpoint:<state:"active"
name:"cali5089b2798cb" profile_ids:"kns.ingress" profile_ids:"ksa.ingress.nginx-ingress-microk8s-serviceaccount" ipv4_nets:"10.1.107.66/32" >
2023-01-25 16:14:58.105 [INFO][96] felix/endpoint_mgr.go 607: Updating per-endpoint chains. id=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"ingress/nginx-ingress-microk8s-controller-7th8s", EndpointId:"eth0"}
2023-01-25 16:14:58.106 [INFO][96] felix/table.go 508: Queueing update of chain. chainName="cali-tw-cali5089b2798cb" ipVersion=0x4 table="filter"
2023-01-25 16:14:58.106 [INFO][96] felix/table.go 508: Queueing update of chain. chainName="cali-fw-cali5089b2798cb" ipVersion=0x4 table="filter"
2023-01-25 16:14:58.106 [INFO][96] felix/endpoint_mgr.go 653: Updating endpoint routes. id=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"ingress/nginx-ingress-microk8s-controller-7th8s", EndpointId:"eth0"}
2023-01-25 16:14:58.106 [INFO][96] felix/endpoint_mgr.go 1220: Applying /proc/sys configuration to interface. ifaceName="cali5089b2798cb"
2023-01-25 16:14:58.106 [INFO][96] felix/endpoint_mgr.go 495: Re-evaluated workload endpoint status adminUp=true failed=false known=true operUp=true status="up" workloadEndpointID=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"ingress/nginx-ingress-microk8s-c
ontroller-7th8s", EndpointId:"eth0"}
2023-01-25 16:14:58.106 [INFO][96] felix/status_combiner.go 58: Storing endpoint status update ipVersion=0x4 status="up" workload=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"ingress/nginx-ingress-microk8s-controller-7th8s", EndpointId:"eth0"}
2023-01-25 16:14:58.114 [INFO][96] felix/status_combiner.go 81: Endpoint up for at least one IP version id=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"ingress/nginx-ingress-microk8s-controller-7th8s", EndpointId:"eth0"} ipVersion=0x4 status="up"
2023-01-25 16:14:58.115 [INFO][96] felix/status_combiner.go 98: Reporting combined status. id=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"ingress/nginx-ingress-microk8s-controller-7th8s", EndpointId:"eth0"} status="up"
2023-01-25 16:14:58.989 [INFO][96] felix/route_table.go 953: Deleting from expected targets cidr=10.1.107.64/26 ifaceName="*NoOIF*" ipVersion=0x4 tableIndex=0
2023-01-25 16:14:58.989 [INFO][96] felix/route_table.go 959: No pending target update, adding back in as an update cidr=10.1.107.64/26 ifaceName="*NoOIF*" ipVersion=0x4 tableIndex=0
2023-01-25 16:14:59.369 [ERROR][93] monitor-addresses/startup.go 925: Unable to query node configuration error=Get "https://10.152.183.1:443/api/v1/nodes?limit=500": dial tcp 10.152.183.1:443: i/o timeout
2023-01-25 16:14:59.369 [WARNING][93] monitor-addresses/utils.go 49: Terminating
2023-01-25 16:14:59.423 [INFO][51074] monitor-addresses/startup.go 427: Early log level set to info
2023-01-25 16:14:59.425 [INFO][51074] monitor-addresses/utils.go 127: Using NODENAME environment for node name mipnyc1
2023-01-25 16:14:59.425 [INFO][51074] monitor-addresses/utils.go 139: Determined node name: mipnyc1
2023-01-25 16:15:23.242 [INFO][96] felix/summary.go 100: Summarising 12 dataplane reconciliation loops over 1m2.6s: avg=8ms longest=30ms ()
2023-01-25 16:15:28.082 [INFO][96] felix/calc_graph.go 460: Local endpoint updated id=WorkloadEndpoint(node=mipnyc1, orchestrator=k8s, workload=ingress/nginx-ingress-microk8s-controller-7th8s, name=eth0)
2023-01-25 16:15:28.082 [INFO][96] felix/int_dataplane.go 1585: Received *proto.WorkloadEndpointUpdate update from calculation graph msg=id:<orchestrator_id:"k8s" workload_id:"ingress/nginx-ingress-microk8s-controller-7th8s" endpoint_id:"eth0" > endpoint:<state:"active"
name:"cali5089b2798cb" profile_ids:"kns.ingress" profile_ids:"ksa.ingress.nginx-ingress-microk8s-serviceaccount" ipv4_nets:"10.1.107.66/32" >
2023-01-25 16:15:28.082 [INFO][96] felix/endpoint_mgr.go 607: Updating per-endpoint chains. id=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"ingress/nginx-ingress-microk8s-controller-7th8s", EndpointId:"eth0"}
2023-01-25 16:15:28.082 [INFO][96] felix/table.go 508: Queueing update of chain. chainName="cali-tw-cali5089b2798cb" ipVersion=0x4 table="filter"
2023-01-25 16:15:28.082 [INFO][96] felix/table.go 508: Queueing update of chain. chainName="cali-fw-cali5089b2798cb" ipVersion=0x4 table="filter"
```

In the beginning the `10.86.14.*** `IP range is used but then it's switched to `10.152.183.1` and it starts failing.

**Comment by ThanKarab** [comment]
So, I managed to find that the problem was the wrong network interface IP in the master node.

I needed to change the `--advertise-address` in the `/var/snap/microk8s/current/args/kube-apiserver` and that fixes the problem after a while.

Currently the `microk8s kubectl get endpoints` return:
```
NAME                           ENDPOINTS                                                          AGE
kubernetes                     10.86.13.12:16443,10.86.14.10:16443                                65m
```

The calico pods in the remote nodes, outside the main cluster, need some time to "sync" and still throw a lot of errors before "finding" the correct IP.

Is there a way to remove the previous endpoint and provide only the IP that is advertised?
